lib/route/nh.c

```diff
@@ -757,6 +840,26 @@ static int nexthop_msg_parser(struct nl_cache_ops *ops, struct sockaddr_nl *who,
 		nexthop->ce_mask |= NH_ATTR_GROUP_TYPE;
 	}
 
+	if (tb[NHA_ENCAP] && tb[NHA_ENCAP_TYPE]) {
+		int err2;
+		struct rtnl_nh_encap *nh_encap = NULL;
+
+		err2 = nh_encap_parse_msg(tb[NHA_ENCAP], tb[NHA_ENCAP_TYPE],
+					  &nh_encap);
+		if (err2 == 0) {
+			err2 = rtnl_nh_set_encap(nexthop, nh_encap);
+			if (err2 < 0) {
+				rtnl_nh_encap_free(nh_encap);
```

Comment:
Must avoid the double-free after `rtnl_nh_set_encap()`.

should also use `_nl_auto_rtnl_nh_encap` and `_nl_steal_pointer()`.

In practice, there is no bug here because `nh_encap_parse_msg()` returns a `nh_encap` if-and-only-if it succeeds. But optimally, the code should be resilient against `nh_encap_parse_msg()` returning an error while also returning a `nh_encap` (e.g. if the implementation is more sloppy and assigns the output `nh_encap` first before determining there is an error). In that case, the code here should just cleanup the returned `nh_encap`. We get that nicely for free with `_nl_auto_rtnl_nh_encap`.